src/librados/IoCtxImpl.cc

```diff
@@ -1655,9 +1646,13 @@ int librados::IoCtxImpl::watch(const object_t& oid, uint64_t *handle,
 
   Objecter::LingerOp *linger_op = objecter->linger_register(oid, oloc, 0);
   *handle = linger_op->get_cookie();
-  linger_op->watch_context = new WatchInfo(this,
-					   oid, ctx, ctx2, internal);
-
+  auto wi = new WatchInfo(this, oid, ctx, ctx2, internal);
+  linger_op->user_data =
+    std::unique_ptr<void, fu2::unique_function<void(void*)>>(
+      wi, [] (void* v) {
+	    delete static_cast<WatchInfo*>(v);
```

Comment:
Ok, +1. Since its used multiple times, it may be worth considering making this a concrete function.

Reply:
DONE. (Also lets us use a function pointer instead of a type-erased function wrapper.)

THOUGH for full generality a std::any might be better.

Reply:
+1, but now the signature can be changified! ;-)